modules/dnn/test/test_misc.cpp

```diff
@@ -93,6 +93,42 @@ TEST(blobFromImageWithParams_4ch, NHWC_scalar_scale)
     }
 }
 
+TEST(blobFromImageWithParams_CustomPadding, letter_box)
+{
+    Mat img(40, 20, CV_8UC4, Scalar(0, 1, 2, 3));
+
+    // Custom padding value that you have added
+    Scalar customPaddingValue(0, 1, 2, 3); // Example padding value
+
+    // Construct target mat with the expected padding value
+    Mat targetCh[4];
+
+    std::vector<uint8_t> valVec = {1,1,1,1,1, 1,1,1,1,1,1,1,1,1,1, 1,1,1,1,1};
+
+    Mat rowM(1, 20, CV_8UC1, valVec.data());
+    for (int i = 0; i < 4; i++) {
+        targetCh[i] = rowM * i;
+    }
+
+    Mat targetImg;
+    merge(targetCh, 4, targetImg);
+    Size targetSize(20, 20);
+
+    // Set up Image2BlobParams with your new functionality
+    Image2BlobParams param;
+    param.size = targetSize;
+    param.paddingmode = DNN_PMODE_LETTERBOX;
+    param.borderValue = customPaddingValue; // Use your new feature here
+
+    // Create blob with custom padding
+    Mat blob = dnn::blobFromImageWithParams(img, param);
+
+    // Create target blob for comparison
+    Mat targetBlob = dnn::blobFromImage(targetImg, 1.0, targetSize);
+
+    EXPECT_EQ(0, cvtest::norm(targetBlob, blob, NORM_INF));
```

Comment:
Where is the check for `customPaddingValue`?

Reply:
What do you mean? Be more specific please.

Reply:
There is no connection of `customPaddingValue` variable and the rest of test. If I change the value - test fails. I expected something like that:
```cpp
TEST(blobFromImageWithParams_CustomPadding, letter_box)
{
    Mat img(40, 20, CV_8UC4, Scalar(0, 1, 2, 3));

    // Custom padding value that you have added
    Scalar customPaddingValue(0, 1, 2, 3); // Example padding value

    Size targetSize(20, 20);

    Mat targetImg = img.clone();
    cv::copyMakeBorder(targetImg, targetImg, 0, 0, 0, targetSize.width, BORDER_CONSTANT, customPaddingValue);

    // Set up Image2BlobParams with your new functionality
    Image2BlobParams param;
    param.size = targetSize;
    param.paddingmode = DNN_PMODE_LETTERBOX;
    param.borderValue = customPaddingValue; // Use your new feature here

    // Create blob with custom padding
    Mat blob = dnn::blobFromImageWithParams(img, param);

    // Create target blob for comparison
    Mat targetBlob = dnn::blobFromImage(targetImg, 1.0, targetSize);

    EXPECT_EQ(0, cvtest::norm(targetBlob, blob, NORM_INF));
}
```

Reply:
There is no connection of `customPaddingValue` with the rest of test. If it's changed - test fails. So I recommend something like that:
```cpp
TEST(blobFromImageWithParams_CustomPadding, letter_box)
{
    Mat img(40, 20, CV_8UC4, Scalar(0, 1, 2, 3));

    // Custom padding value that you have added
    Scalar customPaddingValue(0, 1, 2, 3); // Example padding value

    Size targetSize(20, 20);

    Mat targetImg = img.clone();
    cv::copyMakeBorder(targetImg, targetImg, 0, 0, targetSize.width / 2, targetSize.width / 2, BORDER_CONSTANT, customPaddingValue);

    // Set up Image2BlobParams with your new functionality
    Image2BlobParams param;
    param.size = targetSize;
    param.paddingmode = DNN_PMODE_LETTERBOX;
    param.borderValue = customPaddingValue; // Use your new feature here

    // Create blob with custom padding
    Mat blob = dnn::blobFromImageWithParams(img, param);

    // Create target blob for comparison
    Mat targetBlob = dnn::blobFromImage(targetImg, 1.0, targetSize);

    EXPECT_EQ(0, cvtest::norm(targetBlob, blob, NORM_INF));
}
```